plugins/flytekit-flyin/flytekitplugins/flyin/vscode_lib/decorator.py

```diff
@@ -167,63 +172,77 @@ def download_vscode(vscode_config: VscodeConfig):
         execute_command(f"code-server --install-extension {p}")
 
 
-def vscode(
-    _task_function: Optional[Callable] = None,
-    max_idle_seconds: Optional[int] = MAX_IDLE_SECONDS,
-    port: Optional[int] = 8080,
-    enable: Optional[bool] = True,
-    pre_execute: Optional[Callable] = None,
-    post_execute: Optional[Callable] = None,
-    config: Optional[VscodeConfig] = None,
-):
-    """
-    vscode decorator modifies a container to run a VSCode server:
-    1. Overrides the user function with a VSCode setup function.
-    2. Download vscode server and extension from remote to local.
-    3. Launches and monitors the VSCode server.
-    4. Terminates if the server is idle for a set duration.
-
-    Args:
-        _task_function (function, optional): The user function to be decorated. Defaults to None.
-        max_idle_seconds (int, optional): The duration in seconds to live after no activity detected.
-        port (int, optional): The port to be used by the VSCode server. Defaults to 8080.
-        enable (bool, optional): Whether to enable the VSCode decorator. Defaults to True.
-        pre_execute (function, optional): The function to be executed before the vscode setup function.
-        post_execute (function, optional): The function to be executed before the vscode is self-terminated.
-        config (VscodeConfig, optional): VSCode config contains default URLs of the VSCode server and extension remote paths.
-    """
-
-    if config is None:
-        config = VscodeConfig()
-
-    def wrapper(fn):
-        if not enable:
-            return fn
-
-        @wraps(fn)
-        def inner_wrapper(*args, **kwargs):
-            # 0. Executes the pre_execute function if provided.
-            if pre_execute is not None:
-                pre_execute()
-                logger.info("Pre execute function executed successfully!")
-
-            # 1. Downloads the VSCode server from Internet to local.
-            download_vscode(config)
-
-            # 2. Launches and monitors the VSCode server.
-            # Run the function in the background
-            child_process = multiprocessing.Process(
-                target=execute_command, kwargs={"cmd": f"code-server --bind-addr 0.0.0.0:{port} --auth none"}
-            )
-
-            child_process.start()
-            exit_handler(child_process, max_idle_seconds, post_execute)
-
-        return inner_wrapper
-
-    # for the case when the decorator is used without arguments
-    if _task_function is not None:
-        return wrapper(_task_function)
-    # for the case when the decorator is used with arguments
-    else:
-        return wrapper
+VSCODE_TYPE_VALUE = "vscode"
+
+
+class vscode(ClassDecorator):
+    def __init__(
+        self,
+        fn: Optional[Callable] = None,
+        max_idle_seconds: Optional[int] = MAX_IDLE_SECONDS,
+        port: Optional[int] = 8080,
+        enable: Optional[bool] = True,
```

Comment:
if the default value is not None, we don't need to set it optional. https://docs.python.org/3/library/typing.html#typing.Optional

Reply:
done